packages/opentelemetry-api/src/trace/Sampler.ts

```diff
@@ -25,12 +29,19 @@ export interface Sampler {
   /**
    * Checks whether span needs to be created and tracked.
    *
-   * TODO: Consider to add required arguments https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sampling-api.md#shouldsample
    * @param [parentContext] Parent span context. Typically taken from the wire.
    *     Can be null.
    * @returns whether span should be sampled or not.
    */
-  shouldSample(parentContext?: SpanContext): boolean;
+  shouldSample(
```

Comment:
`shouldSample` was previously returning a boolean and name of this function was also intuitive and sounded like a boolean. This is not the case anymore, so I would suggest changing the name of this function into something that better describes what it does. Maybe something like `prepareSamplingResult` or `getSamplingResult` or `makeSamplingResult` or `takeSamplingResult` or even `createSamplingResult` etc. WDYT ?

Reply:
`shouldSample` matches the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample

Reply:
ok it sounds weird but then sry for confusion